### DISCUSSION TITLE
Default shiftKey behavior on transformer

### DISCUSSION THREAD

**Discussion by pro-nav** [konvajs/konva#1445] [Ideas]
I am working on a project where I need the default behavior of transformer to keep proportions while scaling. To achieve this, I am using the `transformer({ keepRatio: true })`. But now in some cases I want users to allow disregarding the proportions when the shift key is held during the transform. But the current behavior of transformer doesn't make this possible. 

Ideally there should be a variable to something like `shiftBehavior` which can be `default`, `inverted` or `none` depending on which the value of the variable [keepProportions](https://github.com/konvajs/konva/blob/578191d5d4755b4b2d1f9490284b32d9953656ac/src/shapes/Transformer.ts#L709) is calculated.

```js
var keepProportions;
switch(this.shiftBehavior){
    case 'inverted':
        keepProportions = this.keepRatio() && !e.shiftKey;
    case 'none':
        keepProportions = this.keepRatio();
    default:
        keepProportions = this.keepRatio() || e.shiftKey;
        break;
}
```

https://github.com/konvajs/konva/blob/578191d5d4755b4b2d1f9490284b32d9953656ac/src/shapes/Transformer.ts#L709

**Comment by lavrton** [comment]
Do you want to make a Pull request?

**Comment by adrientremblay** [comment]
This change has been added btw for anyone who finds this thread
